pkg/apis/extensions/v1beta1/types.go

```diff
@@ -286,6 +290,47 @@ type DeploymentStatus struct {
 
 	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
 	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Conditions represent the latest available observations of a deployment's current state.
+	Conditions []DeploymentCondition `json:"conditions,omitempty"`
+}
+
+type DeploymentConditionType string
+
+// These are valid conditions of a deployment.
+const (
+	// DeploymentProgressing means the deployment is progressing.
+	DeploymentProgressing DeploymentConditionType = "Progressing"
+	// DeploymentComplete means the deployment has completed progressing.
+	DeploymentComplete DeploymentConditionType = "Complete"
+	// DeploymentFailed means the deployment has failed progressing.
+	DeploymentFailed DeploymentConditionType = "Failed"
+)
+
+// DeploymentCondition describes the state of a deployment at a certain point.
+type DeploymentCondition struct {
+	// Type of deployment condition.
+	Type DeploymentConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status"`
+	// LastProbeTime is the last time the condition was checked.
+	LastProbeTime unversioned.Time `json:"lastProbeTime,omitempty"`
```

Comment:
With all the doubts laid out, I'd actually prefer having both, since each of them is there for separate thing and that differentiation does make sense, imho.